src/util.cpp

```diff
@@ -478,7 +478,15 @@ const boost::filesystem::path &ZC_GetParamsDir()
     if (!path.empty())
         return path;
 
-    path = ZC_GetBaseParamsDir();
+    if (mapArgs.count("-paramsdir")) {
+        path = fs::system_complete(mapArgs["-paramsdir"]);
+        if (!fs::is_directory(path)) {
+            path = "";
```

Comment:
no, thats empty string return make sense: because it handled later, and print error here:
https://github.com/zcash/zcash/blob/master/src/bitcoind.cpp#L95-L99
I checked, fs::is_directory() returns false if function parameter is empty string.
So "-datadir /notexists" -> getdatadir returns "" -> is_directory("") returns false -> error+ exit

i would handle the invalid paramsdir same as getdatadir: adding an userfriendly message (like https://github.com/zcash/zcash/blob/master/src/init.cpp#L703-L710) in ZC_LoadParams, and not throwing a exception to the world :)

Reply:
Well, that code is also wrong, because it checks for the existence of a directory at the path given by `GetDataDir(false)`, but then the error message incorrectly assumes that this is the same as `mapArgs["-datadir"]`.

Reply:
Also, exceptions from this code will definitely be handled and their message printed: https://github.com/zcash/zcash/blob/66e39a0dd6ffd77bcbede1943195dd456f859cd6/src/bitcoind.cpp#L170-L174